### ISSUE TITLE
Nested fields in Geo

### ISSUE THREAD

**Issue by domoritz** [vega/vega-lite#3351]
```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v2.1.json",
  "width": 500,
  "height": 500,
  "layer": [
    {
      "data": {
        "url": "https://unpkg.com/world-atlas@1/world/110m.json",
        "format": {
          "type": "topojson",
          "feature": "countries"
        }
      },
      "projection": {
        "type": "orthographic"
      },
      "mark": {
        "type": "geoshape",
        "fill": "mintcream",
        "stroke": "black",
        "strokeWidth": 0.35
      }
    },
    {
      "data": {
        "url": "https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/all_week.geojson",
        "format": {
          "type": "json",
          "property": "features"
        }
      },
      "projection": {
        "type": "orthographic"
      },
      "mark": {
        "type": "circle",
        "opacity": 0.25,
        "fill": "red"
      },
      "encoding": {
        "x": {"field": "geometry.coordinates[0]", "type": "longitude"},
        "y": {"field": "geometry.coordinates[1]", "type": "latitude"},
        "size": {"field": "properties.mag", "type": "quantitative"}
      }
    }
  ]
}
```

**Comment by domoritz** [comment]
I don't know how to fix this properly. We don't track where a field comes from and so it's hard to know whether to remove nesting or not from a field. It would be nice if geopoint transforms could produce nested data but that doesn't seem like a good option either.

Any ideas @kanitw @willium?

**Comment by kanitw** [comment]
I think making geopoint produce nested data is bad too.  

I think we need to design common infrastructure for field tracking.  This is causing problem for `bin` and `timeUnit` transforms already #2862.

**Comment by domoritz** [comment]
Not sure whether we really need to track it or whether we can just always flatten data.

**Comment by kanitw** [comment]
> Not sure whether we really need to track it or whether we can just always flatten data.

That's true. Actually for this issue, I think we should just name the output of geopoint by the unit spec's name (Like `layer_0_geo_point_x/y`)  instead of just adding `_geo` suffix to latitude/longitude fields. 

This will help eliminate misleading output fields too. 

For example, given input
```
"x": {"field": "long", "type": "longitude"},
"y": {"field": "lat", "type": "latitude"},
```

Currently geopoint will output   `long_geo` and `lat_geo`, which are both misleading. 
They are actually projected x- and y-positions, both of which are functions of lat and long.

**Comment by kanitw** [comment]
Fixed in #3498 

![image](https://user-images.githubusercontent.com/111269/37243841-ffc4e400-244d-11e8-871c-456b829e05dc.png)
